kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/vault/KubeVaultTestKmsFacade.java

```diff
@@ -126,11 +133,50 @@ protected URI getVaultUrl() {
         return URI.create("http://" + vault.getVaultUrl());
     }
 
+    /**
+     * Gets vault version.
+     *
+     * @return the vault version
+     */
+    public String getVaultVersion() {
+        return vault.getVersionInstalled();
+    }
+
     @Override
     public TestKekManager getTestKekManager() {
         return new VaultTestKekManager();
     }
 
+    /**
+     * Is correct version installed boolean.
+     *
+     * @return the boolean
+     */
+    public boolean isCorrectVersionInstalled() {
+        String installedVersion = getVaultVersion();
+        String expectedVersion = VaultTestKmsFacade.HASHICORP_VAULT.split(":")[1];
+
+        return compareTo(installedVersion, expectedVersion) == 0;
+    }
+
+    private int compareTo(String currentVersion, String expectedVersion) {
```

Comment:
Don't use the method name compareTo, you risk confusing the reviewer.  They'll likely think `java.lang.Comparable#compareTo`